client/chain-spec/src/chain_spec.rs

```diff
@@ -285,10 +285,12 @@ impl<G, E: serde::de::DeserializeOwned> ChainSpec<G, E> {
 
 	/// Parse json file into a `ChainSpec`
 	pub fn from_json_file(path: PathBuf) -> Result<Self, String> {
-		let file = File::open(&path)
-			.map_err(|e| format!("Error opening spec file `{}`: {}", path.display(), e))?;
+		// We read the entire file into memory first, as this is *a lot* faster than using
+		// `serde_json::from_reader`. See https://github.com/serde-rs/json/issues/160
+		let bytes = std::fs::read(&path).map_err(|e| format!("Error reading spec file: {}", e))?;
```

Comment:
It'd most likely be better to `mmap` the file than to read it whole into memory (especially if it can be a few gigs in size). Could you try mmaping it through the `memmap2` crate (we already use it transitively as a dependency through `parity-db` anyway) instead? (It should be roughly as fast, but at a significantly lower memory usage.)

Reply:
I mean having multiple gigabytes in this file is rather rare, however I'm fine with doing these change if we are going to change this here anyway. @koute could you maybe push the required changes to this pr?

Reply:
Yep, I know. Still, apparently some people do it, as evidenced by this PR. (:

Sure; done!

Reply:
Ty, if you now approve your own changes @koute we should be ready with this pr :D